File: stustapay/core/schema/tse.py
import enum
from typing import Optional

from pydantic import BaseModel


class TseStatus(enum.Enum):
    new = "new"
    active = "active"
    disabled = "disabled"
    failed = "failed"


class TseType(enum.Enum):
    diebold_nixdorf = "diebold_nixdorf"


class UpdateTse(BaseModel):
    name: str
    ws_url: str
    ws_timeout: float
    password: str


class NewTse(UpdateTse):
    type: TseType
    serial: Optional[str]


class Tse(NewTse):
    node_id: int
    id: int
    status: TseStatus
    hashalgo: Optional[str]
    time_format: Optional[str]
    public_key: Optional[str]
    certificate: Optional[str]
    process_data_encoding: Optional[str]

File: web/libs/models/src/lib/tse.ts
import { z } from "zod";

export const UpdateTseSchema = z.object({
  name: z.string().min(1),
  ws_url: z.string().min(1),
  ws_timeout: z.number(),
  password: z.string(),
  first_operation: z.string().nullable(),
});
export type UpdateTse = z.infer<typeof UpdateTseSchema>;

export const NewTseSchema = UpdateTseSchema.merge(
  z.object({
    serial: z.string().min(1),
    type: z.literal("diebold_nixdorf"),
  })
);
export type NewTse = z.infer<typeof NewTseSchema>;
